src/client.cc

```diff
@@ -683,16 +683,10 @@ minio::s3::PutObjectResponse minio::s3::Client::PutObject(PutObjectArgs args) {
         "SSE operation must be performed over a secure connection");
   }
 
-  char* buf = NULL;
-  if (args.part_count > 0) {
-    buf = new char[args.part_size];
-  } else {
-    buf = new char[args.part_size + 1];
-  }
+  std::vector<char> buf(args.part_size + 1);
```

Comment:
This requires good amount of testing with various upload cases. Just stick to `delete []buf`

Reply:
Changed to delete []